test/ctia/task/migration/migrate_es_stores_test.clj

```diff
@@ -44,12 +44,22 @@
                   helpers/fixture-properties:clean
                   es-helpers/fixture-properties:es-store]))
 
-(setup!) ;; init migration conn and properties
-(def es-props (get-in @props/properties [:ctia :store :es]))
-(def es-conn (connect (:default es-props)))
-(def migration-index (get-in es-props [:migration :indexname]))
+;; This a is a `defn` to prevent side-effects at compile time
+(defn es-props []
+  (get-in @props/properties [:ctia :store :es]))
+
+;; This a is a `defn` to prevent side-effects at compile time
+(defn es-conn []
+  (connect (:default es-props)))
+
+;; This a is a `defn` to prevent side-effects at compile time
+(defn migration-index []
+  (get-in es-props [:migration :indexname]))
 
 (defn fixture-clean-migration [t]
+  (defonce setup ;; ugly, but must be done in order to prevent an indefinitely blocking call (which can affect code reloading, or re-running this ns's tests)
+    (setup!) ;; init migration conn and properties
```

Comment:
This solution is not ideal (but not worse than the current one)

Ideally we'd teardown the setup (i.e. stop the underlying http server). I wasn't able to figure out how.

Reply:
Perhaps `shutdown-ctia!`? But I really don't know if it's designed for iteration in tests.

How about something like:
```clojure
(defonce setup nil)
(defn fixture-clean-fixture [t]
  (alter-var-root! #'setup #(or % (setup!)))
  ...)
```
Or maybe just add a `declare` for the var to keep the ns graph stable.